pkg/util/namespace/namespace.go

```diff
@@ -23,3 +23,70 @@ func IsOpenShiftSystemNamespace(ns string) bool {
 		strings.HasPrefix(ns, "kube-") ||
 		strings.HasPrefix(ns, "openshift-")
 }
+
+// FilteredOpenShiftNamespace returns true if ns is a namespace in the defined hardcoded map
+func FilteredOpenShiftNamespace(ns string) bool {
```

Comment:
As I understand it, this function was created because the prefix `openshift-` here https://github.com/Azure/ARO-RP/pull/1994/files#diff-e6fbfa5cca3e92386cc833c67dea826373fc746278df21f8e778d13d9ed6b103R24 can capture customer `openshift-*` namespaces such as `openshift-gitops`.

Does it make sense to fix the previous function, or add an entirely new one? As I saw in a previous review, this new function is not actually being called or performing any work at this point. What was the plan with this moving forward?

If we just patched the functionality of `IsOpenShiftSystemNamespace` to include each namespace we care about granularly this new function would not be required.

Reply:
Hi Ben, thanks for your detailed review feedback. 

Based on the discussion with @m1kola in the [ADO - 12682885](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/12682885). - He suggested we would go with hard-coding for now. Extracting namespaces from an OCP release could be a next iteration. 

That's why I just added a new function "FilteredOpenShiftNamespace" and still keep the previous functions at this stage.

Reply:
Sorry if it wasn't clear from the work item, but the idea is to replace existing filtering with new, more accurate. Also take a look at the acceptance creteria of the work item: this PR doesn't meet it as we are not using the new filtering at the moment.

Reply:
Originally there were 2 distinct functions, so the name `FilteredOpenShiftNamespace` sort of made sense, but now that it has replaced `IsOpenShift` and `IsOpenShiftSystemNamespace` entirely I would suggest that the function name should be changed across the board to `IsOpenShiftNamespace`.